numpy/_core/src/umath/ufunc_type_resolution.c

```diff
@@ -1115,12 +1115,46 @@ PyUFunc_MultiplicationTypeResolver(PyUFuncObject *ufunc,
     type_num2 = PyArray_DESCR(operands[1])->type_num;
 
     /* Use the default when datetime and timedelta are not involved */
-    if (!PyTypeNum_ISDATETIME(type_num1) && !PyTypeNum_ISDATETIME(type_num2)) {
+    if (!PyTypeNum_ISDATETIME(type_num1) && !PyTypeNum_ISDATETIME(type_num2)
+        && !((PyTypeNum_ISSTRING(type_num1) && PyTypeNum_ISINTEGER(type_num2))
+             || (PyTypeNum_ISINTEGER(type_num1) && PyTypeNum_ISSTRING(type_num2)))) {
         return PyUFunc_SimpleUniformOperationTypeResolver(ufunc, casting,
                     operands, type_tup, out_dtypes);
     }
 
-    if (type_num1 == NPY_TIMEDELTA) {
+    if (PyTypeNum_ISSTRING(type_num1) || PyTypeNum_ISSTRING(type_num2)) {
+        // This is wrong, but only the DType matters here (String or Unicode) and
```

Comment:
not immediately clear to me what's wrong about this, can you clarify a little?

Reply:
Oh this is probably not the right place for this comment. This was related to setting the output dtype to the string input's dtype, which will obviously have the wrong `elsize`. I've updated the comment.